sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -113,6 +114,25 @@ public class CryptographyAsyncClient {
             throw new IllegalArgumentException("The JSON Web Key's key type property is not configured.");
         }
 
+        if (!isCurveSupportedByRuntimeJavaVersion(jsonWebKey.getCurveName())) {
+            String javaVersion = System.getProperty("java.version");
```

Comment:
A lot of string parsing for versions (really? no `Version` type of 3-4 numeric components?). Maybe cache the result statically so you don't have to do this ever time? Not like the runtime is going to change out from a running process.

Reply:
Good idea, I'll cache it to avoid check every time.